### DISCUSSION TITLE
Socket io emit using Access token

### DISCUSSION THREAD

**Discussion by sree9578** [miguelgrinberg/Flask-SocketIO#1824] [Q&A] [answered]
Hi @miguelgrinberg ,
    Is it possible to emit using access token for specific users instead of session id?

**Comment by miguelgrinberg** [comment, marked as answer]
Yes, it is possible to do it, but you have to implement the mapping between access tokens and session IDs in your application. This package only accepts the session IDs when referring to clients.

**Reply by sree9578** [reply]
Can u guide me to some documentation for session IDs and access token mapping please @miguelgrinberg

**Reply by sree9578** [reply]
I need to emit messages from api calls..
i am able to get only access token in api calls 
i can get session id from socket.on()  
i can't map the session id to this access token ..in order to emit message to a specific user

**Reply by miguelgrinberg** [reply]
You can create a mapping between users and session ids in your connect and disconnect handlers. In the connect handler you add the user to the map, in the disconnect you remove it. This map (which can be implemented as a Python dict) will allow you to find a given user and obtain their session id.

**Reply by sree9578** [reply]
Thanks @miguelgrinberg  it was helpful
